tests/dtm/test_rgw_s3_process_restart.py

```diff
@@ -0,0 +1,287 @@
+                                                      que, self.test_cfg['size']))
+        proc_write_op.start()
+
+        self.log.info("Step 2: Perform rgw_s3 Process Restart for %s times During Read "
+                      "Operations", DTM_CFG["rgw_restart_cnt"])
+        resp = self.dtm_obj.process_restart(master_node=self.master_node_list[0],
+                                            health_obj=self.health_obj,
+                                            pod_prefix=const.SERVER_POD_NAME_PREFIX,
+                                            container_prefix=const.RGW_CONTAINER_NAME,
+                                            process=self.rgw_process, check_proc_state=False,
+                                            restart_cnt=DTM_CFG["rgw_restart_cnt"])
+        assert_utils.assert_true(resp, "Failure observed during process restart/recovery")
+
+        self.log.info("Step 2: Wait for WRITE Operation to complete.")
+        if proc_write_op.is_alive():
+            proc_write_op.join()
+        resp = que.get()
+        assert_utils.assert_true(resp[0], resp[1])
+        workload_info = resp[1]
+
+        self.log.info("Step 3: Perform READ Operation on data written in Step 1")
+        self.dtm_obj.perform_ops(workload_info, que, False, True, True)
+        resp = que.get()
+        assert_utils.assert_true(resp[0], resp[1])
+
+        self.test_completed = True
+
+        self.log.info("ENDED: Verify WRITEs during rgw_s3 restart using pkill")
+
+    @pytest.mark.lc
+    @pytest.mark.dtm
+    @pytest.mark.tags("TEST-42246")
+    def test_delete_during_rgw_svc_restart(self):
+        """Verify DELETEs during rgw_s3 service restart using pkill."""
+        self.log.info("STARTED: Verify DELETEs during rgw_s3 service restart using pkill")
+        test_prefix = 'test-42246'
+        wr_output = Queue()
+        test_cfg = DTM_CFG["test_42246"]
+
+        event = threading.Event()  # Event to be used to send intimation of rgw_s3 process restart
+
+        self.log.info("Step 1: Perform write Operations :")
+        self.dtm_obj.perform_write_op(bucket_prefix=f"bucket-{test_prefix}",
+                                      object_prefix=f"object-{test_prefix}",
+                                      no_of_clients=self.test_cfg['clients'],
+                                      no_of_samples=self.test_cfg['samples'],
+                                      log_file_prefix=test_prefix, queue=wr_output,
+                                      loop=test_cfg['num_loop'])
+        resp = wr_output.get()
+        assert_utils.assert_true(resp[0], resp[1])
+
+        buckets = self.s3_test_obj.bucket_list()[1]
+        self.log.info("Step 1: Successfully created %s buckets & performed WRITEs with variable "
+                      "size objects.", len(buckets))
+
+        output = Queue()
+        self.log.info("Step 2: Start Continuous DELETEs of buckets %s in background", buckets)
```

Comment:
so we are deleting all buckets? Can we delete some and list remaining after restart?

Reply:
That we are planning to take in [TEST-42249](https://jts.seagate.com/browse/TEST-42249)[TEST-42249](https://jts.seagate.com/browse/TEST-42249)

Reply:
That we are planning to take in [TEST-42249](https://jts.seagate.com/browse/TEST-42249)[TEST-42249](https://jts.seagate.com/browse/TEST-42249)

Reply:
ok